kafka-connect-s3/src/main/java/io/confluent/connect/s3/storage/S3Storage.java

```diff
@@ -60,14 +62,20 @@ public S3Storage(S3SinkConnectorConfig conf, String url) {
   }
 
   public AmazonS3 newS3Client(S3SinkConnectorConfig config) {
+    String proxyHost = config.getString(S3_PROXY_HOST);
+    int proxyPort = config.getInt(S3_PROXY_PORT);
+
+    ClientConfiguration clientConfiguration = PredefinedClientConfigurations.defaultConfig();
+    clientConfiguration.withUserAgentPrefix(String.format(VERSION_FORMAT, Version.getVersion()));
+    if(proxyHost != null && !proxyHost.isEmpty()) {
```

Comment:
Right, that's where I'm getting at. If you do `withProxyHost(null)`, aws sdk will eventually check the appropriate system property when it'll use this config (for instance `https.proxyHost`). The check we add here for proxyHost being not null is kind of a no-op.

Here's the code that is used internally to determine whether a proxy will be used: 
https://github.com/aws/aws-sdk-java/blob/master/aws-java-sdk-core/src/main/java/com/amazonaws/http/apache/client/impl/ApacheHttpClientFactory.java#L116 

aws sdk will check the system properties anyways, even if you don't call `setProxyHost` or `withProxyHost`. That also means that setting a proxy is possible even with the current config (although I'm totally up for adding this configuration in S3 Connector's properties as well).